flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/expressions/ApiExpressionUtils.java

```diff
@@ -78,12 +83,12 @@ public static SymbolExpression symbol(TableSymbol symbol) {
 		return new SymbolExpression(symbol);
 	}
 
-	public static UnresolvedFieldReferenceExpression unresolvedFieldRef(String name) {
-		return new UnresolvedFieldReferenceExpression(name);
+	public static UnresolvedReferenceExpression unresolvedFieldRef(String name) {
+		return new UnresolvedReferenceExpression(name);
 	}
 
 	public static TableReferenceExpression tableRef(String name, Table table) {
```

Comment:
I agree it is worth unifying java and scala, but in this case I think it is better to aim for the scala behavior. Mostly because the table might not necessarily be registered in the catalog (might be just an "in place" query), in this case we cannot resolve the table by name.

Reply:
True, registering the table under the hood is not perfect design.

Reply:
The `table.toString` will register the table, so both cases will register the table. However, I think it's ok if you think we should aim for the scala behavior.

Reply:
Yes, maybe we can make this nicer once we rework table environments and how the interplay with catalogs.